client/components/status/status.js

```diff
@@ -293,7 +283,18 @@ export default class Status extends React.Component {
         if (settings.mobile_phone_verification) {
           userInfo.phone_number = phone_number;
         }
-        this.setState({username, password, is_active, is_verified, userInfo});
+        this.setState(
+          {username, password, is_active, is_verified, userInfo},
+          () => {
+            // if the user is being automatically logged in but it's not
+            // active anymore (eg: has been banned)
+            // automatically perform log out
+            setIsActive(is_active);
+            if (!is_active) {
```

Comment:
just to be extra sure, I would do `is_active === false`. I'll try that now and if works will push it here.